packages/tools/pie-wrapper-react/scripts/add-react-wrapper.js

```diff
@@ -81,27 +81,33 @@ export function addReactWrapper (customElementsObject, folderName = process.argv
         components.forEach((component) => {
             const events = getEvents(component.class);
 
+            // Prepare the event names and comments
+            let eventNames = '';
+            if (component.class.events && component.class.events.length > 0) {
+                eventNames = events?.flat().reduce((pre, event) => {
```

Comment:
```suggestion
            if (component.class?.events?.length) {
                eventNames = events?.flat()?.reduce((pre, event) => {
```

Reply:
Just chatted in person but for others viewing this, I think it would be preferable for us to avoid optional chaining and instead allow this process to error and break if there is any missing required data. This way we are less likely to introduce a possibility of silently failing when building the wrappers.